### PR TITLE
Implement Test.todo

### PR DESCRIPTION
**Pull request by mgold**
Related to #94, but first: how do we implement the "don't show these failures in the output" behavior?

* Filter them in `Test.Runner.run`. This ensures that the node and HTML runners are consistent, and that this PR can be merged without dependencies on other repos.
* Filter them in the runners. This seems like the more "correct" thing to do because it allows, for example, the node runner to take a flag to show even (or only) the TODO test failures. (`--todo=(show|hide|only), default: hide`)

Either way, we need to resolve this before merging, or at least releasing.

@rtfeldman I'd be happy to take a crack at adding support to the node runner, just let me know.

### CODE REVIEW

File: src/Test.elm [hunk 1]
```diff
@@ -99,6 +100,34 @@ test desc thunk =
...
+but you can't implement them all at once. When you replace `todo` with a real
+test, you'll be able to see if it fails without clutter from tests still not
+implemented. But, unlike leaving yourself comments, you'll be prompted to
+implement these tests because your suite will fail.
+
+    describe "a new thing"
+        [ todo "does what is expected in the common case"
+        , todo "correctly handles an edge case I just thought of"
+        ]
+-}
```
Comment (soenkehahn) [inline src/Test.elm]: I think it would be a good idea to mention two things:

- `todo` tests are usually not meant to be committed to a code-base. It's only a tool for development.
- `todo` tests are sometimes called `pending` in other test-frameworks. (This comment would also be helpful for people searching for the word `pending` in the docs or on google.)

**Comment by rtfeldman** [comment]
@mgold I agree that the runners should be in charge of filtering these.

That does mean we'd need to expose something like `isTodo : Test -> Bool` so the runners can find out which tests are `TODO`.

**Comment by mgold** [comment]
@rtfeldman Seem very one-off. Is there some way to incorporate this into a much larger export of information about failures, without exposing internal types?

**Comment by rtfeldman** [comment]
> Seem very one-off

Agreed, but to be fair, we haven't wanted anything else yet!

It's arguably better to do it this way until we end up with several of these and want to combine them. Supposing in the future we wanted to add a check for (let's say) whether it's a fuzz test, adding `isFuzz : Test -> Bool` is a non-breaking change, whereas modifying some `TestInfo` record would be a breaking change.

**Comment by mgold** [comment]
Okay. The other thing is that this is going into the Runner module, which is "experts only" of sorts, so I'm more okay with the prospect of accruing cruft there than elsewhere.

> we haven't wanted anything else yet

I understand if you haven't found the time to work on the runner, but don't we want to expose the detailed `Reason` info for highlighting?

**Comment by mgold** [comment]
> this is going into the Runner module

Whelp, `getFailure` is in the Expect module. It really shouldn't be because it's concerned with inspecting tests, not writing them. I'll still put `isTodo` in `Test.Runner` and make a note to move `getFailure` whenever we have a major change.

**Comment by rtfeldman** [comment]
Go ahead and move it. I think we're going to want a major change anyway, because I have a ton of improvements coming down the pipe for `node-test-runner` (including @stoeffel's doctests!) so I think now is a good time to do breaking changes like this and https://github.com/elm-community/elm-test/issues/95

**Comment by mgold** [comment]
Okay! I'll pull this issue into the next release with the label.

**Comment by mgold** [comment]
Okay, new version is up with `Test.Runner.todo`, tests for that, and `Expect.getFailure` moved over to `Test.Runner`.

**Comment by rtfeldman** [comment]
Love it! 😻

Merge when ready @mgold!

**Comment by soenkehahn** [comment]
Awesome! @rtfeldman and @mgold: Thanks a lot. Can't wait to get my hands on it.
